### ISSUE TITLE
Uber and UberEats Crashes

### ISSUE THREAD

**Issue by dylangerdaly** [microg/GmsCore#1945]
**Describe the bug**
Both Uber and UberEats crash on start

**To Reproduce**
1. Open Uber
2. Observe Crash

**Expected behavior**
It not crash

**Screenshots**
N/A

**System**
Android Version: 13
Custom ROM: N/A

**Additional context**
Crash Log

```
05-30 17:18:57.955 12768 12768 D AndroidRuntime: Shutting down VM
05-30 17:18:57.955 12768 12768 E AndroidRuntime: FATAL EXCEPTION: main
05-30 17:18:57.955 12768 12768 E AndroidRuntime: Process: com.ubercab, PID: 12768
05-30 17:18:57.955 12768 12768 E AndroidRuntime: java.lang.ClassCastException: java.lang.SecurityException cannot be cast to com.google.android.gms.common.api.c
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at ffb.e.a(SourceFile:126)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at ffb.e.lambda$j8g2WqGn9D1I2VnkFBXKA8HJ_NU6(Unknown Source:0)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at ffb.-$$Lambda$e$j8g2WqGn9D1I2VnkFBXKA8HJ_NU6.onFailure(Unknown Source:4)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at kx.v.run(SourceFile:4)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:942)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at kp.b.dispatchMessage(SourceFile:6)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7903)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
05-30 17:18:57.955 12768 12768 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
05-30 17:18:57.956 12768 12928 E ActivityThread: Failed to find provider info for com.uber.companion
```

**Comment by mar-v-in** [comment]
This should've been fixed with https://github.com/microg/GmsCore/commit/33e24696ff5e06c18851465b63c19bd634581238 (also available as https://microg.org/dl/core-nightly.apk). Can you confirm?

**Comment by dylangerdaly** [comment]
I fixed this by granting both applications Location Permissions, if you don't grant these applications the Location Permission then I observe a crash

**Comment by mar-v-in** [comment]
To be precise: I can't reproduce crashes with current master anymore, even if location permission is revoked.

**Comment by dylangerdaly** [comment]
> This should've been fixed with [33e2469](https://github.com/microg/GmsCore/commit/33e24696ff5e06c18851465b63c19bd634581238) (also available as https://microg.org/dl/core-nightly.apk). Can you confirm?

Apologies, I didn't read this, indeed that commit fixed Uber and UberEats when location isn't granted, thank you

**Comment by robinpaulson** [comment]
Would this fix assist withe bug in #1953?
